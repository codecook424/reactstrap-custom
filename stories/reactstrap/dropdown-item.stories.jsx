import * as React from "react";
import {
  Dropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
} from "reactstrap";

let sharedBooleanDefaults = {
  type: "boolean",
  defaultValue: false,
  table: { defaultValue: { summary: false } },
};

let propDescriptions = {
  active:
    "Convey the active state to assistive technologies. Use the aria-current attribute â€” using the page value for the current page, or true for the current item in a set.",
  header: "Label sections of actions in any dropdown menu.",
  divider: "Mark visible items as disabled.",
  disabled: "Indicade that available options are disabled.",
  text: "Place any freeform text within a dropdown menu with text and use spacing utilities.",
};

export default {
  title: "Components/DropdownItem",
  component: DropdownItem,
  render: Template,
  args: { children: "A dropdown item" },
  argTypes: {
    active: {
      ...sharedBooleanDefaults,
      description: propDescriptions.active,
    },
    children: {
      type: "string",
      description: "Element content (React.node)",
      defaultValue: "A dropdown item",
      table: { defaultValue: { summary: undefined } },
    },
    divider: {
      ...sharedBooleanDefaults,
      description: propDescriptions.divider,
    },
    disabled: {
      ...sharedBooleanDefaults,
      description: propDescriptions.disabled,
    },
    header: {
      ...sharedBooleanDefaults,
      description: propDescriptions.header,
    },
    text: {
      ...sharedBooleanDefaults,
      description: propDescriptions.text,
    },
  },
  parameters: {
    docs: {
      description: {
        component:
          "An universal element for adding items, groupings, and dividers to dropdown menus.",
      },
    },
  },
  decorators: [withMinHeight],
};

export const Default = {};

export const Active = {
  args: {
    active: true,
    children: "An active dropdown item",
    "aria-current": true,
  },
  ...addDocsDescription(propDescriptions.active),
};

export const Disabled = {
  args: {
    disabled: true,
    children: "A disabled dropdown item",
  },
  ...addDocsDescription(propDescriptions.disabled),
};

export const Header = {
  args: {
    header: true,
    children: "A header",
  },
  ...addDocsDescription(propDescriptions.header),
};

export const Text = {
  args: {
    text: true,
    children: "A text dropdown item (can't click)",
  },
  ...addDocsDescription(propDescriptions.text),
};

export const Divider = {
  args: {
    divider: true,
    children: "A text dropdown item (can't click)",
  },
  ...addDocsDescription(propDescriptions.divider),
};

// private

function addDocsDescription(description) {
  return {
    parameters: {
      docs: {
        description: {
          story: description,
        },
      },
    },
  };
}

function Template(args) {
  return (
    <div>
      <DropdownItem {...args} />
      <DropdownItem>A dropdown item</DropdownItem>
    </div>
  );
}

function withMinHeight(Story) {
  function DemoLayout(props) {
    return <div style={{ minHeight: "5rem" }} {...props} />;
  }

  return <DemoLayout>{Story()}</DemoLayout>;
}
