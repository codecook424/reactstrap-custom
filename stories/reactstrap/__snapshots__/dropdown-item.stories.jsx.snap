// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/DropdownItem Active smoke-test 1`] = `
<div style="min-height: 5rem;">
  <div>
    <button type="button"
            aria-current="true"
            tabindex="0"
            role="menuitem"
            class="dropdown-item active"
    >
      An active dropdown item
    </button>
    <button type="button"
            tabindex="0"
            role="menuitem"
            class="dropdown-item"
    >
      A dropdown item
    </button>
  </div>
</div>
`;

exports[`Components/DropdownItem Default smoke-test 1`] = `
<div style="min-height: 5rem;">
  <div>
    <button type="button"
            tabindex="0"
            role="menuitem"
            class="dropdown-item"
    >
      A dropdown item
    </button>
    <button type="button"
            tabindex="0"
            role="menuitem"
            class="dropdown-item"
    >
      A dropdown item
    </button>
  </div>
</div>
`;

exports[`Components/DropdownItem Disabled smoke-test 1`] = `
<div style="min-height: 5rem;">
  <div>
    <button type="button"
            disabled
            tabindex="-1"
            class="disabled dropdown-item"
    >
      A disabled dropdown item
    </button>
    <button type="button"
            tabindex="0"
            role="menuitem"
            class="dropdown-item"
    >
      A dropdown item
    </button>
  </div>
</div>
`;

exports[`Components/DropdownItem Divider smoke-test 1`] = `
<div style="min-height: 5rem;">
  <div>
    <div tabindex="-1"
         class="dropdown-divider"
    >
      A text dropdown item (can't click)
    </div>
    <button type="button"
            tabindex="0"
            role="menuitem"
            class="dropdown-item"
    >
      A dropdown item
    </button>
  </div>
</div>
`;

exports[`Components/DropdownItem Header smoke-test 1`] = `
<div style="min-height: 5rem;">
  <div>
    <h6 tabindex="-1"
        class="dropdown-header"
    >
      A header
    </h6>
    <button type="button"
            tabindex="0"
            role="menuitem"
            class="dropdown-item"
    >
      A dropdown item
    </button>
  </div>
</div>
`;

exports[`Components/DropdownItem Text smoke-test 1`] = `
<div style="min-height: 5rem;">
  <div>
    <span tabindex="-1"
          class="dropdown-item-text"
    >
      A text dropdown item (can't click)
    </span>
    <button type="button"
            tabindex="0"
            role="menuitem"
            class="dropdown-item"
    >
      A dropdown item
    </button>
  </div>
</div>
`;
